File: test_data/code/java/1656-design-an-ordered-stream.java
import java.util.*;

class OrderedStream {
    private Map<Integer, String> map;
    private List<String> chunk;

    public OrderedStream(int n) {
        map = new HashMap<>();
        chunk = new ArrayList<>();
        for (int i = 1; i <= n; i++) {
            map.put(i, null);
        }
    }

    public List<String> insert(int idKey, String value) {
        if (idKey > map.size()) {
            return chunk;
        }
        if (map.get(idKey) == null) {
            map.put(idKey, value);
            while (chunk.size() < idKey && map.containsKey(idKey)) {
                chunk.add(map.remove(idKey));
            }
        }
        return chunk;
    }
}

File: test_data/code/python/1656-design-an-ordered-stream.py
class OrderedStream:

    def __init__(self, n: int):
        self.stream = [""] * (n + 1)
        

    def insert(self, idKey: int, value: str) -> List[str]:
        self.stream[idKey] = value
        result = []
        while self.stream[idKey] != "":
            result.append(self.stream[idKey])
            idKey += 1
        return result